admin/cluster_setup.py

```diff
@@ -240,3 +246,46 @@ def main(reactor, args, base_path, top_level):
                 print("Be sure to preserve the required files.")
 
     raise SystemExit(result)
+
+
+def _make_client(reactor, cluster):
+    """
+    Create a :class:`FlockerClient` object for accessing the given cluster.
+
+    :param reactor: The reactor.
+    :param flocker.provision._common.Cluster cluster: The target cluster.
+    :return: The client object.
+    :rtype: flocker.apiclient.FlockerClient
+    """
+    control_node = cluster.control_node.address
+    certificates_path = cluster.certificates_path
+    cluster_cert = certificates_path.child(b"cluster.crt")
+    user_cert = certificates_path.child(b"user.crt")
+    user_key = certificates_path.child(b"user.key")
+    return FlockerClient(reactor, control_node, REST_API_PORT,
+                         cluster_cert, user_cert, user_key)
+
+
+def _wait_for_nodes(client, count):
+    """
+    Wait until nodes join the cluster.
+
+    :param flocker.apiclient.FlockerClient client: The client connected to
+        the cluster (its control node).
+    :param int count: The expected number of nodes in the cluster.
+    :return: ``Deferred`` firing when the number of nodes in the cluster
+        reaches the target.
+    """
+    def got_all_nodes():
+        d = client.list_nodes()
+        d.addErrback(write_failure, logger=None)
+
+        def check_node_count(nodes):
+            print("Waiting for nodes, "
+                  "got {} out of {}".format(len(nodes), count))
+            return len(nodes) == count
+
+        d.addCallback(check_node_count)
+        return d
+
+    return loop_until(client._reactor, got_all_nodes, repeat(1, 120))
```

Comment:
I'm starting to cringe at `loop_until` uses because `loop_until` fails somewhat opaquely.  This isn't really a problem with this PR, more a problem with the timeout behavior of `loop_until`.  Maybe I should go file an issue about that ...

So I guess this comment isn't actionable.